src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -1305,22 +1274,21 @@ bool ColorControlServer::moveSaturationCommand(app::CommandHandler * commandObj,
     // New command.  Need to stop any active transitions.
     stopAllColorTransitions(endpoint);
 
+    // now, kick off the state machine.
+    initSaturationTransitionState(endpoint, colorSaturationTransitionState);
+
     if (moveMode == EMBER_ZCL_SATURATION_MOVE_MODE_STOP || rate == 0)
     {
+        // Per spec any hue transition must also be cancelled.
```

Comment:
Is that true for the rate = 0 case?

Reply:
No you are correct, I overlooked that. In fact, the implemented behaviour is wrong when `rate == 0`

`MoveMode field is set to 1 (up) or 3 (down) and the Rate field has a value of zero, the command has no effect and a response command SHALL be sent in response, with the status code set to INVALID_COMMAND.`

Will fix that in a bit and verify it that fix applies to other commands